Patch largely based on cwmbuild.sh from http://tamentis.com/hacks/cwm/
written by Bertrand Janin, updated by Christian Neukirchen.

diff -aur cwm_build.old/calmwm.h cwm_build/calmwm.h
--- cwm_build.old/calmwm.h	2010-01-29 09:56:14.069865288 +0100
+++ cwm_build/calmwm.h	2010-01-29 09:57:50.840856134 +0100
@@ -21,6 +21,10 @@
 #ifndef _CALMWM_H_
 #define _CALMWM_H_
 
+/* ugly stuff */
+#define TAILQ_END(head) NULL
+#define __dead
+
 #include <X11/Xatom.h>
 #include <X11/Xft/Xft.h>
 #include <X11/Xlib.h>
diff -aur cwm_build.old/includes.h cwm_build/includes.h
--- cwm_build/includes.h        1969-12-31 19:00:00.000000000 -0500
+++ cwm_build.new/includes.h    2009-08-30 19:20:14.000000000 -0400
@@ -0,0 +1 @@
+
diff -aur cwm_build.old/Makefile cwm_build/Makefile
--- cwm_build.old/Makefile	2010-01-29 09:56:14.071106346 +0100
+++ cwm_build/Makefile	2010-01-29 10:04:04.889862860 +0100
@@ -1,14 +1,23 @@
 # $OpenBSD: Makefile,v 1.15 2009/01/19 20:23:19 todd Exp $
 
-.include <bsd.xconf.mk>
+#.include <bsd.xconf.mk>
 
 PROG=		cwm
 
+BINDIR=		/usr/bin
+
 SRCS=		calmwm.c screen.c xmalloc.c client.c menu.c \
 		search.c util.c xutil.c conf.c input.c xevents.c group.c \
 		kbfunc.c mousefunc.c font.c parse.y
 
-CPPFLAGS+=	-I${X11BASE}/include -I${X11BASE}/include/freetype2 -I${.CURDIR}
+OBJS=		calmwm.o screen.o xmalloc.o client.o menu.o \
+		search.o util.o xutil.o conf.o input.o xevents.o group.o \
+		kbfunc.o mousefunc.o font.o strlcpy.o strlcat.o y.tab.o \
+		strtonum.o fgetln.o
+
+X11BASE=	/usr
+
+CPPFLAGS+=	-I${X11BASE}/include -I${X11BASE}/include/freetype2 -I.
 
 CFLAGS+=	-Wall
 
@@ -20,7 +29,23 @@
 
 CLEANFILES=	cwm.cat1 cwmrc.cat5
 
-obj: _xenocara_obj
 
-.include <bsd.prog.mk>
-.include <bsd.xorg.mk>
+all: $(PROG)
+
+y.tab.c: parse.y
+	byacc parse.y
+
+
+$(PROG): $(OBJS) y.tab.o
+	$(CC) $(OBJS) ${LDADD} -o ${PROG}
+
+$(OBJS): %.o: %.c
+	$(CC) -c $(CFLAGS) $(CPPFLAGS) $<
+
+install: ${PROG}
+	install -m 755 cwm /usr/local/bin/
+	install -m 644 cwm.1 /usr/local/man/man1
+	install -m 644 cwmrc.5 /usr/local/man/man5
+
+#.include <bsd.prog.mk>
+#.include <bsd.xorg.mk>
diff -aur cwm_build.old/xevents.c cwm_build/xevents.c
--- cwm_build.old/xevents.c	2010-01-29 09:56:14.069865288 +0100
+++ cwm_build/xevents.c	2010-01-29 10:01:43.472994946 +0100
@@ -29,6 +29,7 @@
 
 #include <err.h>
 #include <errno.h>
+#include <signal.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
